daq/acl_state_collector.py

```diff
@@ -0,0 +1,70 @@
+"""Process ACL states"""
+
+from __future__ import absolute_import
+
+import logger
+
+LOGGER = logger.get_logger('acl')
+
+
+class AclStateCollector:
+    """Processing ACL states for ACL counting"""
+
+    def __init__(self):
+        self._switch_configs = {}
+
+    def get_port_acl_count(self, switch, port, port_acl_samples):
+        """Return the ACL count for a port"""
+
+        switch_config = self._switch_configs.get(switch)
+        if not switch_config:
+            LOGGER.warning('Switch not defined in Faucet dps config: %s', switch)
+            return {}
+
+        port_config = switch_config.ports.get(port)
+        if not port_config:
+            LOGGER.warning('Port not defined in Faucet dps config: %s:%s', switch, port)
+            return {}
+
+        acls_config = port_config.acls_in
+        if not acls_config:
+            LOGGER.warning('No ACLs applied to port: %s:%s', switch, port)
+            return {}
+
+        return self._get_port_rules_count(switch, port, acls_config[0], port_acl_samples)
+
+    # pylint: disable=protected-access
+    def _get_port_rules_count(self, switch, port, acl_config, acl_samples):
```

Comment:
is acl_samples == port_acl samples?  Unless there's a good reason, use consistent variable name.

Reply:
Updated so that acl_samples is used in all the places.